### DISCUSSION TITLE
How to limit the size of request headers?

### DISCUSSION THREAD

**Discussion by lithbitren** [hyperium/hyper#3696] [Q&A] [answered]
Some malicious requests from scripts have very huge header sizes. How to set a limit or check these header sizes? It would be great to quickly end the TCP connection and get the IP address to add it to an IP block list.

**Comment by seanmonstar** [comment, marked as answer]
The limit that currently exists is the [max_buf_size](https://docs.rs/hyper/latest/hyper/server/conn/http1/struct.Builder.html#method.max_buf_size). If headers are not "complete" within the max size, then an error is returned and the connection closed.

A potential new builder option is planned that would allow setting a header size limit smaller than the max buf size, but does not exist at the moment.
